From 11b923c99cbe3580885ce40c322277fc823107a0 Mon Sep 17 00:00:00 2001
From: Naveen Saini <naveen.kumar.saini@intel.com>
Date: Tue, 2 Feb 2021 13:39:53 +0800
Subject: [PATCH] IGC/VectorCompiler/CMakeLists.txt: link to external
 LLVMGenXIntrinsics

By default LLVMGenXIntrinsics is to be build In-tree, but we want to
link externally.

Fix llvm-tblgen path.

Upstream-Status: Inappropriate [configuration specific]

Signed-off-by: Naveen Saini <naveen.kumar.saini@intel.com>
---
 IGC/VectorCompiler/CMakeLists.txt | 32 ++-----------------------------
 1 file changed, 2 insertions(+), 30 deletions(-)

diff --git a/IGC/VectorCompiler/CMakeLists.txt b/IGC/VectorCompiler/CMakeLists.txt
index 86f343ee..b4268160 100644
--- a/IGC/VectorCompiler/CMakeLists.txt
+++ b/IGC/VectorCompiler/CMakeLists.txt
@@ -53,10 +53,7 @@ if(IGC_BUILD__USING_SYSTEM_LLVM OR (WIN32 AND LLVM_USE_PREBUILT))
   message(STATUS "[VC] Using system llvm")
 
   # Need to search for llvm-tblgen
-  find_program(LLVM_TABLEGEN_EXE "llvm-tblgen"
-    ${LLVM_TOOLS_BINARY_DIR}
-    NO_DEFAULT_PATH
-    )
+  find_program(LLVM_TABLEGEN_EXE "llvm-tblgen")
   if(LLVM_TABLEGEN_EXE-NOTFOUND)
     message(FATAL_ERROR "[VC] llvm-tblgen is not found")
   endif()
@@ -140,32 +137,7 @@ if(LLVM_ON_WIN32)
   add_compile_options(/wd4141)
 endif()
 
-if(DEFINED VC_INTRINSICS_SRC)
-  set(INTRSRC "${VC_INTRINSICS_SRC}/GenXIntrinsics")
-endif()
-
-if(NOT DEFINED INTRSRC)
-  set(INTRSRC "${CMAKE_CURRENT_SOURCE_DIR}/../../../vc-intrinsics/GenXIntrinsics")
-endif()
-
-message(STATUS "[VC] Using vc-intrinsics source from: ${INTRSRC}")
-
-# We are using prebuilt SPIRV and building intrinsics.
-set(INTRBUILD "${CMAKE_CURRENT_BINARY_DIR}/intrbuild")
-add_subdirectory(${INTRSRC} ${INTRBUILD})
-# Make separate target for intrinstics headers.
-# Ideally, this should be inside intrinsics, but
-# this breaks in-tree build of intrinsics with LLVM.
-add_library(VCIntrinsicsHeaders INTERFACE)
-target_include_directories(VCIntrinsicsHeaders
-  INTERFACE
-  ${INTRSRC}/include
-  ${INTRBUILD}/include
-  )
-add_dependencies(VCIntrinsicsHeaders
-  GenXIntrinsics
-  )
-
+find_package(LLVMGenXIntrinsics REQUIRED)
 include(cmake/spirv.cmake)
 include(${IGC_SOURCE_DIR}/cmake/utils.cmake)
 
-- 
2.17.1

